judgels-backends/judgels-commons/judgels-persistence-core/src/main/java/judgels/persistence/SearchOptions.java

```diff
@@ -0,0 +1,13 @@
+package judgels.persistence;
+
+import java.util.Map;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface SearchOptions {
+    Map<String, String> getTerms();
+
+    class Builder extends ImmutableSearchOptions.Builder {
```

Comment:
Just omit the empty constructor.

```suggestion
    class Builder extends ImmutableSearchOptions.Builder {}
```